pymchelper/executor/runner.py

```diff
@@ -45,26 +51,51 @@ def __init__(self, jobs=None, keep_workspace_after_run=False, output_directory='
         self.workspace_manager = WorkspaceManager(output_directory=output_directory,
                                                   keep_workspace_after_run=keep_workspace_after_run)
 
-    def run(self, settings):
+    def run(self):
         """
         Execute parallel simulation processes, creating workspace (and working directories) in the `output_directory`
         In case of successful execution return True, otherwise return False
         """
         start_time = timeit.default_timer()
 
-        # SHIELD-HIT12A and Fluka require RNG seeds to be integers greater or equal to 1
-        # each of the workers needs to have its own different RNG seed
-        rng_seeds = range(1, self.jobs + 1)
+        if self.settings.simulator_type in [SimulatorType.shieldhit, SimulatorType.fluka]:
+            # SHIELD-HIT12A and Fluka require RNG seeds to be integers greater or equal to 1
+            # each of the workers needs to have its own different RNG seed
+            rng_seeds = range(1, self.jobs + 1)
+
+        elif self.settings.simulator_type == SimulatorType.topas:
+            # for topas we don't need to create multiple working directories and a pool of workers
+            # as we can use embedded parallelization in topas
+            # for that we need to modify the input file and set the number of threads to the number of jobs
+            # we set one rng seed, so one working directory and one worker will be created
+
+            modified_input_path = str(self.settings.input_path).replace(".txt", "_modified.txt")
```

Comment:
Make `input_path` a Pathlib object and use `with_suffix`

Reply:
I think it's not possible to change the suffix to `"_modified.txt"` using `with_suffix` method - when I use it like this: `Path(self.settings.input_path).with_suffix("_modified.txt")` it returns an error: 
`ValueError: Invalid suffix '_modified.txt'`

Reply:
Thats correct. I as more concern about the fact that `.txt` may be part of directory filename (I've such cases....).
What about replacing core filename (stem?) using pathlib ?

Reply:
Something like this?
`modified_input_filename = Path(self.settings.input_path).name.replace(".txt", "_modified.txt")
 modified_input_path = Path(self.settings.input_path).parent / modified_input_filename
`